Lib/dataclasses.py

```diff
@@ -222,6 +222,29 @@ def __repr__(self):
 # https://bugs.python.org/issue33453 for details.
 _MODULE_IDENTIFIER_RE = re.compile(r'^(?:\s*(\w+)\s*\.)?\s*(\w+)')
 
+# Types for which deepcopy(obj) is known to return obj unmodified
+# Used to skip deepcopy in asdict and astuple for performance
+_ATOMIC_TYPES = {
+    # Common JSON Serializable types
+    types.NoneType,
+    bool,
+    int,
+    float,
+    complex,
+    bytes,
+    str,
+    # Other types that are also unaffected by deepcopy
+    types.EllipsisType,
+    types.NotImplementedType,
+    types.CodeType,
+    types.BuiltinFunctionType,
+    types.FunctionType,
+    type,
+    range,
+    property,
+    # weakref.ref,  # weakref is not currently imported by dataclasses directly
+}
```

Comment:
Nit: could be a frozenset:

```suggestion
_ATOMIC_TYPES = frozenset({
    # Common JSON Serializable types
    types.NoneType,
    bool,
    int,
    float,
    complex,
    bytes,
    str,
    # Other types that are also unaffected by deepcopy
    types.EllipsisType,
    types.NotImplementedType,
    types.CodeType,
    types.BuiltinFunctionType,
    types.FunctionType,
    type,
    range,
    property,
    # weakref.ref,  # weakref is not currently imported by dataclasses directly
})
```

Reply:
Sorry, I've done this but because I'd already used one of the other suggestions github blocked accepting this change directly.

Reply:
Yeah I spotted, no worries :)